### ISSUE TITLE
Nullability on Task.Result with Tuples has unexpected warnings

### ISSUE THREAD

**Issue by glatzert** [dotnet/roslyn#45338]
**Version Used**: 
Microsoft Visual Studio Enterprise 2019
Version 16.6.2
VisualStudio.16.Release/16.6.2+30204.135

C# Tools   3.6.0-4.20251.5+910223b64f108fcf039012e0849befb46ace6e66
IntelliCode Extension   1.0

**Steps to Reproduce**:

1. Create method with signature `public Task<(int? count, object? error)> Method()`
2. Use Task.FromResult(x) with x from `(0, (object)null)`, `((int?)null, new object())`, `(1, new object())`
3. See warnings.

**Expected Behavior**:
I'd expect those cases to not issue a warning, since the nullability of the method allows all combinations shown.

**Actual Behavior**:
Warnings are issued.

**Comment by jaredpar** [comment]
Can you provide a full sample here?

**Comment by glatzert** [comment]
Sure thing:

```c-sharp
using System.Threading.Tasks;

namespace ClassLibrary1
{
    public class Class1
    {
        public Task<(bool isValid, object? validationErrors)> ValidateAsync()
        {
            return Task.FromResult((true, (object)null)); //This will warn
        }
    }
}
```

**Comment by glatzert** [comment]
Disregard the last one - sorry:

```c-sharp
public class Class1
    {
        public Task<(object? obj1, object? obj2)> ValidateAsync()
        {
            var result = (new object(), (object?)null);
            return Task.FromResult(result);
        }
    }
```

**Comment by jaredpar** [comment]
This behavior is "By Design". 

Yes it is legal for `(object, object?)` to assign to `(object?, object?)`. That conversion though does not hold when the tuple is the type parameter of the `Task<T>` type. In that case the tuple must match exactly.
